test/components/views/room_settings/RoomProfileSettings-test.tsx

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2024 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from "react";
+import { render, screen, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { mocked } from "jest-mock";
+import { EventType, MatrixClient, MatrixEvent, Room } from "matrix-js-sdk/src/matrix";
+
+import { mkStubRoom, stubClient } from "../../../test-utils";
+import RoomProfileSettings from "../../../../src/components/views/room_settings/RoomProfileSettings";
+
+const BASE64_GIF = "R0lGODlhAQABAAAAACw=";
+const AVATAR_FILE = new File([Uint8Array.from(atob(BASE64_GIF), (c) => c.charCodeAt(0))], "avatar.gif", {
+    type: "image/gif",
+});
+
+const ROOM_ID = "!floob:itty";
+
+describe("RoomProfileSetting", () => {
+    let client: MatrixClient;
+    let room: Room;
+
+    beforeEach(() => {
+        client = stubClient();
+        room = mkStubRoom(ROOM_ID, "Test room", client);
+    });
+
+    it("uploads a room avatar", async () => {
```

Comment:
```suggestion
    it("handles uploading a room avatar", async () => {
```

... might be a more accurate representation of this test. Likewise the others.